src/server/pbsd_main.c

```diff
@@ -2045,53 +2047,56 @@ main(int argc, char **argv)
 				clear_exec_vnode();
 				first_run = 0;
 			}
-
-			/* if time or event says to run scheduler, do it */
-
-			/* if we have a high prio sched command, send it 1st */
-			if (svr_do_sched_high != SCH_SCHEDULE_NULL)
-				schedule_high();
-
-
-			if (svr_do_schedule == SCH_SCHEDULE_RESTART_CYCLE) {
-
-				/* send only to existing connection */
-				/* since it is for interrupting current */
-				/* cycle */
-				/* NOTE: both primary and secondary scheduler */
-				/* connect must have been setup to be valid */
-				if ((scheduler_sock2 != -1) &&
-					(scheduler_sock != -1)) {
-					if (put_sched_cmd(scheduler_sock2,
-						svr_do_schedule, NULL) == 0) {
-						log_event(PBSEVENT_DEBUG2,
+			for (psched = (pbs_sched*) GET_NEXT(svr_allscheds); psched; psched = (pbs_sched*) GET_NEXT(psched->sc_link)) {
+				/* if time or event says to run scheduler, do it */
+
+				/* if we have a high prio sched command, send it 1st */
+				if ((strcmp(psched->sch_attr[SCHED_ATR_sched_state].at_val.at_str, SC_DOWN)) &&
+					psched->svr_do_sched_high != SCH_SCHEDULE_NULL)
+					schedule_high(psched);
+
+
+				if (psched->svr_do_schedule == SCH_SCHEDULE_RESTART_CYCLE) {
```

Comment:
do we need to check for scheduler state as down here?

Reply:
This additional check is not required and I removed it.